src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs

```diff
@@ -406,6 +406,8 @@ private void topControl_MouseLeftButtonDown(object sender, MouseButtonEventArgs
         {
             if (ViewModel == null || Keyboard.Modifiers == System.Windows.Input.ModifierKeys.Control) return;
 
+            ViewModel.OnRequestsSelection(this, EventArgs.Empty);
```

Comment:
can you remind me why we created this event, and why we are raising it here?

Reply:
this appears to be part of one of my multiple attempts to solve the pinning problem, I just removed this.

Reply:
@SHKnudsen, I just revisited this and it appears that we added `ViewModel.OnRequestsSelection(this, EventArgs.Empty);` because the selection is done directly in the code behind, so that means that when we subscribe to the `OnRequestsSelection` it never runs because it was never executed. Im replacing this with the `SelectCommand.Execute` which I think should be the way of doing it. Let me know if this is ok.

Reply:
Ah i remember this event now, think we added this because we need someway of knowing the node had been selected before the selection actually happened, so that still makes sense IMO, we should add some comments about why we are doing it though.

> Im replacing this with the SelectCommand.Execute which I think should be the way of doing it

Not sure it understand this?

Reply:
I was not sure on calling  `ViewModel.OnRequestsSelection(this, EventArgs.Empty)` directly, 
![image](https://user-images.githubusercontent.com/25138291/124266446-ea406780-db2e-11eb-8296-4bc9757a0d36.png)

and then this event triggers this method on the code behind:
![image](https://user-images.githubusercontent.com/25138291/124266751-44412d00-db2f-11eb-8985-18e2337a0955.png)

or doing it through the ViewModel.SelectCommand,

![image](https://user-images.githubusercontent.com/25138291/124266125-78681e00-db2e-11eb-8372-bd08abf8ab59.png)

 which is a command that is never used, but uses the Select and CanSelect Methods, which may be used in the future to determine if a node can be selected.

![image](https://user-images.githubusercontent.com/25138291/124265988-52db1480-db2e-11eb-91d7-e785476914eb.png)

and then of calling the select:

![image](https://user-images.githubusercontent.com/25138291/124266298-b107f780-db2e-11eb-8791-35957f762d70.png)

and then calls the `OnRequestsSelection`
![image](https://user-images.githubusercontent.com/25138291/124266382-d432a700-db2e-11eb-803e-933cb0eea14d.png)

and then this event triggers this method on the code behind:
![image](https://user-images.githubusercontent.com/25138291/124266751-44412d00-db2f-11eb-8985-18e2337a0955.png)

haha its a little bit tricky to follow the logic behind selection

Reply:
yea this might need a call for some more explanation 😅